source/implementations/f7/Meadow.F7/F7PlatformOS.cs

```diff
@@ -92,4 +92,52 @@ public int[] GetProcessorUtilization()
     {
         return new[] {100 - Core.Interop.Nuttx.meadow_idle_monitor_get_value()};
     }
+
+
+    /// <summary>
+    /// Enum representing different server certificate validation error returns.
+    /// </summary>
+    public enum ServerCertificateValidationError
+    {
+        InvalidMode = -1,
+        CannotChangeAfterInitialization = -2
+    }
+
+    /// <inheritdoc/>
+    public void SetServerCertificateValidationMode(ServerCertificateValidationMode authmode)
+    {
+        Resolver.Log.Trace($"Attempting to set the server certificate validation mode to {authmode}");
+
+        int authModeInt = (int)authmode;
+        if (authModeInt < 0 || authModeInt > 2)
```

Comment:
Use Enum.GetNames().Length or add a terminal “max” enum value, instead of hardcoding 2